src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -319,8 +320,13 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                         $params = array_merge($params, $data[$paramName]);
                     }
                 } elseif ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
-                    $params[] = $data[$key];
-                    // don't run set for a parameter passed to the constructor
+                    $parameterData = $data[$key];
+                    if (null !== $constructorParameter->getClass()) {
+                        $parameterData = $this->serializer->denormalize($parameterData, $constructorParameter->getClass()->getName(), null, $context);
```

Comment:
@dunglas I'm doing a `denormalize` but the normalizer is... `SerializerAware`, not `NormalizerAware`. This works because of the way the serializer is setup but it's violating the interfaces. What should we do about it? Calling `deserializer` even though it will be more expensive?